### PR TITLE
fix `try-runtime fast-forward` when run from the substrate cli and add test

### PR DESCRIPTION
**Pull request by liamaharon**
Closes: https://github.com/paritytech/substrate/issues/13831

## Description

`fast-forward` requires a custom `BlockBuildingInfoProvider` with the blocktime set (see discussion here https://github.com/paritytech/substrate/pull/12896).

The custom provider had been set for the `node-template` cli, but not the `substrate` cli.

- [x] Use a custom `BlockBuildingInfoProvider` with the `substrate` cli
- [ ] Add clearly labelled block building info providers for `node-template`, `substrate`, `polkadot`, `kusama`
- [ ] Allow the block building info provider to be configured by cli args
- [x] Test that `follow-chain` exits with a success status code, and perform a basic check on the output

### CODE REVIEW

File: test-utils/cli/src/lib.rs [hunk 1]
```diff
@@ -61,10 +61,10 @@ use tokio::io::{AsyncBufReadExt, AsyncRead};
...
 /// [`Child`]: std::process::Child
 pub fn start_node() -> Child {
-	Command::new(cargo_bin("substrate"))
+	Command::new(cargo_bin("node-template"))
 		.stdout(process::Stdio::piped())
 		.stderr(process::Stdio::piped())
-		.args(&["--dev", "--tmp", "--ws-port=45789", "--no-hardware-benchmarks"])
+		.args(&["--dev", "--ws-port=45789"])
 		.spawn()
 		.unwrap()
```
Comment (bkchr) [inline test-utils/cli/src/lib.rs:69]: Why this change?

File: utils/frame/try-runtime/cli/tests/fast_forward.rs [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+#![cfg(unix)]
+
+#[cfg(feature = "try-runtime")]
+mod tests {
+	use assert_cmd::cargo::cargo_bin;
+	use regex::Regex;
+	use std::{
+		process::{self},
```
Comment (bkchr) [inline utils/frame/try-runtime/cli/tests/fast_forward.rs:26]:
```suggestion
		process,
```

File: utils/frame/try-runtime/cli/tests/fast_forward.rs [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+	use substrate_cli_test_utils as common;
+	use tokio::process::Command;
+
+	#[tokio::test]
+	async fn follow_chain_works() {
+		// Build substrate so binaries used in the test use the latest code.
+		common::build_substrate(&["--features=try-runtime"]);
+
+		let n_blocks = 10;
+		common::run_with_timeout(Duration::from_secs(60), async move {
```
Comment (bkchr) [inline utils/frame/try-runtime/cli/tests/fast_forward.rs:39]:
```suggestion
		common::run_with_timeout(Duration::from_secs(10 * 60), async move {
```

60 seconds is way too low for CI.

File: test-utils/cli/src/lib.rs (hunk 1 above)
Comment (liamaharon) [reply test-utils/cli/src/lib.rs:69]: Just to make things simpler. 

`--dev` makes `--tmp` redundant, and `node-template` doesn't run any hardware benchmarks by default. 

Let me know if I'm missing something though.

**Comment by liamaharon** [comment]
I'm converting this PR back to a draft with the intention of getting `fast-forward` configurable to work with `node-template`, `substrate`, `polkadot` and `kusama` before reopening for review.

File: bin/node/cli/src/command.rs [hunk 4]
```diff
@@ -32,6 +32,9 @@ use sp_keyring::Sr25519Keyring;
 
 use std::sync::Arc;
 
+#[cfg(feature = "try-runtime")]
+use try_runtime_cli::block_building_info::timestamp_with_aura_info;
```
Comment (kianenigma) [inline bin/node/cli/src/command.rs:36]: I personally always prefer using the full path where needed, or doing the import more locally, rather than feature gated imports, but up to you.

**Comment by liamaharon** [comment]
Closing this for now, need to solve issue with per-chain inherents first.
